c++/src/kj/compat/http.c++

```diff
@@ -1110,6 +1112,150 @@ kj::String HttpHeaders::toString() const {
   return serialize(nullptr, nullptr, nullptr, nullptr);
 }
 
+// -----------------------------------------------------------------------------
+
+
+namespace {
+
+// RFC9110 section 14.1: https://www.rfc-editor.org/rfc/rfc9110#section-14.1
+
+static bool consumeByteRangeUnit(const char*& ptr) {
+  const char* p = ptr;
+  p = skipSpace(p);
+
+  // Match case-insensitive "bytes"
+  if (*p != 'b' && *p != 'B') return false;
+  if (*(++p) != 'y' && *p != 'Y') return false;
+  if (*(++p) != 't' && *p != 'T') return false;
+  if (*(++p) != 'e' && *p != 'E') return false;
+  if (*(++p) != 's' && *p != 'S') return false;
+  ++p;
+
+  p = skipSpace(p);
+  ptr = p;
+  return true;
+}
+
+enum HttpRangeParseError {
+  INVALID, // Parsing failed
+  UNSATISFIABLE // Able to parse and pointer advanced, incorrect for another reason
```

Comment:
Below, it looks like this error code is used to mean "syntactically valid, but out-of-range", e.g. the range exceeds `Content-Length`.

Perhaps it would be clearer to separate the bounds-checking code from the parsing code? Parse the values as they appear first, then check the result against the content-length.

Then the functions could just return `Maybe` which is perhaps a bit clearer.